autovalet_navigation/scripts/autovalet_navigation/move_base_listener.py

```diff
@@ -0,0 +1,200 @@
+
+    def goalCallback(self,goal_msg):
+        
+
+        self.plan_state = State.Planning
+
+        self.goal_pose = goal_msg
+
+        self.start_pose = getCurrentPose()
+
+        if self.debug:
+            self.printState()
+
+            rospy.logwarn("Goal:")
+            rospy.logwarn("Position")
+            rospy.logwarn("  x: %f",self.goal_pose.pose.position.x)
+            rospy.logwarn("  y: %f",self.goal_pose.pose.position.y)
+            rospy.logwarn("  z: %f",self.goal_pose.pose.position.z)
+            rospy.logwarn("Orientation");
+            rospy.logwarn("  x: %f",self.goal_pose.pose.orientation.x)
+            rospy.logwarn("  y: %f",self.goal_pose.pose.orientation.y)
+            rospy.logwarn("  z: %f",self.goal_pose.pose.orientation.z)
+            rospy.logwarn("  w: %f",self.goal_pose.pose.orientation.w)
+
+    def MBstatusCallback(self,status_array_msg):
+
+        # print(status_array_msg.__dict__)
+        ACTIVE = 1
+        SUCCEEDED = 3
+        ABORTED = 4
+
+        if status_array_msg.status_list:
+            current_status = status_array_msg.status_list[-1]
+
+            if current_status.status != ACTIVE and self.plan_state == State.Waiting:
+                return
+
+            if current_status.status == ACTIVE:
+                if self.plan_state == State.Waiting:
+                    self.plan_state = State.Planning
+            
+                elif self.plan_state == State.Planning:
+                    current_pose = getCurrentPose()
+
+                    if not arePosesClose(current_pose,self.start_pose):
+                        self.plan_state = State.Executing
+                    else:
+                        return
+                elif self.plan_state == State.Executing:
+                    return
+
+            elif current_status.status == SUCCEEDED:
+                if self.plan_state == State.Waiting:
+                    return
+                else:
+                    self.plan_state = State.Success
+                    if self.debug:
+                        self.printState()
+                    self.plan_state = State.Waiting
+            
+            elif current_status.status == ABORTED and self.plan_state != State.Waiting:
+                if self.plan_state == State.Waiting:
+                    return
+
+                else:
+                    self.plan_state = State.Fail
+                    if self.debug:
+                        self.printState()
+                    self.plan_state = State.Waiting
+
+        else:
+            return
+
+        if self.debug:
+            self.printState()
+
+
+# # class AutoValet:
+
+# #     def __init__():
+
+# #         self.lane_detector = LaneDetector # import from eva's pkg
+
+# #         self.next_pose = None
+
+# #         self.parking_pose = None
+
+# #         self.pub = rospy.Publisher("/move_base_simple/goal", PoseStamped, queue_size=1)
+
+
+
+if __name__ == '__main__':
+
+    # AV = AutoValet()
```

Comment:
clean up commented lines maybe? I believe we won't ever need them since you create this class in the state machine.

Reply:
done